File: PRACTICA/POLIMORFISMO/ejercicioP1/videojuego.py
#1. Sea la clase Videojuego:
class VideoJuego:
    def __init__(self, nombre="Candy", plataforma="app", cantidad_jugadores=2):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidad_jugadores = cantidad_jugadores

    def mostrar(self):
        print(f"{self.nombre} {self.plataforma} {self.cantidad_jugadores}")

    def agregar_jugadores(self, cantidad=1):
        self.cantidad_jugadores += cantidad

class VideoJuegoOnline(VideoJuego):
    def agregar_jugadores(self, cantidad=1):
        print("(Online) Jugadores conectados...")
        self.cantidad_jugadores += cantidad * 2  

class VideoJuegoLocal(VideoJuego):
    def agregar_jugadores(self, cantidad=1):
        print("(Local) Agregando jugadores en la misma consola...")
        self.cantidad_jugadores += cantidad


j1 = VideoJuego()
j2 = VideoJuegoLocal("FIFA", "PlayStation", 4)
j3 = VideoJuegoOnline("Minecraft", "PC", 1)

# Mostrar
print("---Videojuegos---")
j1.mostrar()
j2.mostrar()
j3.mostrar()

print("\n--- Agregar jugadores  ---")
j1.agregar_jugadores()
j1.mostrar()

j2.agregar_jugadores(2)
j2.mostrar()

j3.agregar_jugadores(2)
j3.mostrar()

File: PRACTICA/POLIMORFISMO/ejercicioP1/ejercicio5/VideoJuego.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ejercicio5;

/**
 *
 * @author Administrador
 */
public class VideoJuego {
    private String nombre,plataforma;
    private int cantidadJugadores;
    public VideoJuego(){
        nombre="Candy";
        plataforma="app";
        cantidadJugadores=2;
    }
    //b) Sobrecargar el constructor 2 veces
     public VideoJuego(String plataforma) {
        this.nombre = "Minecraft";
        this.plataforma = plataforma;
        this.cantidadJugadores =1;
    }
    public VideoJuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }
    //c) Implementar un método mostrar()
    public void mostrar(){
        System.out.println(nombre+" "+plataforma+" "+cantidadJugadores);
    }
    //Sobrecargar el método agregarJugadores() donde en el primero se agregue solo 1 jugador y en otro se ingrese una cantidad de jugadores a aumentar.
    public void agregarJugadores(){
         this.cantidadJugadores += 1;
    }
    public void agregarJugadores(int cantidad) {
        this.cantidadJugadores += cantidad;
    }
        
}
